json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+class WPCOM_JSON_API_Get_Option_Endpoint extends Jetpack_JSON_API_Endpoint {
+
+	protected $needed_capabilities = 'manage_options';
+
+	public $option_name;
+	public $site_option;
+
+	function result() {
+		if ( $this->site_option ) {
+			return array( 'option_value' => get_site_option( $this->option_name ) );
+		}
+		return array( 'option_value' => get_option( $this->option_name ) );
+	}
+
+	function validate_input( $object ) {
+		$query_args = $this->query_args();
+		$this->option_name = isset( $query_args['option_name'] ) ? $query_args['option_name'] : false;
+		if ( ! $this->option_name ) {
+			return new WP_Error( 'option_name_not_set', __( 'You must specify an option_name', 'jetpack' ) );
+		}
+		$this->site_option = isset( $query_args['site_option'] ) ? $query_args['site_option'] : false;
+		if ( ! in_array( $this->option_name, $this->get_readable_options( $this->site_option ) ) ) {
+			return new WP_Error( 'option_name_not_in_whitelist', __( 'You must specify a whitelisted option_name', 'jetpack' ) );
+		}
+		return true;
+	}
+
+	function get_readable_options( $site_option = false ) {
+		if ( $site_option ) {
+			/**
+			 * Filter the list of site options that are readable from the API.
+			 *
```

Comment:
Could you add the module name as well, like here?
https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php#L36

Thanks!